File: tests/algorithms/x/Java/strings/is_pangram.java
public class Main {
    static String s1;
    static String s2;

    static boolean is_pangram(String input_str) {
        String[] letters = ((String[])(new String[]{}));
        int i = 0;
        while (i < _runeLen(input_str)) {
            String c = input_str.substring(i, i+1).toLowerCase();
            boolean is_new = !(java.util.Arrays.asList(letters).contains(c));
            if (!(c.equals(" ")) && ("a".compareTo(c) <= 0) && (c.compareTo("z") <= 0) && is_new) {
                letters = ((String[])(java.util.stream.Stream.concat(java.util.Arrays.stream(letters), java.util.stream.Stream.of(c)).toArray(String[]::new)));
            }
            i = i + 1;
        }
        return letters.length == 26;
    }

    static boolean is_pangram_faster(String input_str) {
        String alphabet = "abcdefghijklmnopqrstuvwxyz";
        boolean[] flag = ((boolean[])(new boolean[]{}));
        int i_1 = 0;
        while (i_1 < 26) {
            flag = ((boolean[])(appendBool(flag, false)));
            i_1 = i_1 + 1;
        }
        int j = 0;
        while (j < _runeLen(input_str)) {
            String c_1 = input_str.substring(j, j+1).toLowerCase();
            int k = 0;
            while (k < 26) {
                if ((alphabet.substring(k, k+1).equals(c_1))) {
flag[k] = true;
                    break;
                }
                k = k + 1;
            }
            j = j + 1;
        }
        int t = 0;
        while (t < 26) {
            if (!(Boolean)flag[t]) {
                return false;
            }
            t = t + 1;
        }
        return true;
    }

    static boolean is_pangram_fastest(String input_str) {
        String s = input_str.toLowerCase();
        String alphabet_1 = "abcdefghijklmnopqrstuvwxyz";
        int i_2 = 0;
        while (i_2 < _runeLen(alphabet_1)) {
            String letter = alphabet_1.substring(i_2, i_2+1);
            if (!(s.contains(letter))) {
                return false;
            }
            i_2 = i_2 + 1;
        }
        return true;
    }
    public static void main(String[] args) {
        {
            long _benchStart = _now();
            long _benchMem = _mem();
            s1 = "The quick brown fox jumps over the lazy dog";
            s2 = "My name is Unknown";
            System.out.println(_p(is_pangram(s1)));
            System.out.println(_p(is_pangram(s2)));
            System.out.println(_p(is_pangram_faster(s1)));
            System.out.println(_p(is_pangram_faster(s2)));
            System.out.println(_p(is_pangram_fastest(s1)));
            System.out.println(_p(is_pangram_fastest(s2)));
            long _benchDuration = _now() - _benchStart;
            long _benchMemory = _mem() - _benchMem;
            System.out.println("{");
            System.out.println("  \"duration_us\": " + _benchDuration + ",");
            System.out.println("  \"memory_bytes\": " + _benchMemory + ",");
            System.out.println("  \"name\": \"main\"");
            System.out.println("}");
            return;
        }
    }

    static boolean _nowSeeded = false;
    static int _nowSeed;
    static int _now() {
        if (!_nowSeeded) {
            String s = System.getenv("MOCHI_NOW_SEED");
            if (s != null && !s.isEmpty()) {
                try { _nowSeed = Integer.parseInt(s); _nowSeeded = true; } catch (Exception e) {}
            }
        }
        if (_nowSeeded) {
            _nowSeed = (int)((_nowSeed * 1664525L + 1013904223) % 2147483647);
            return _nowSeed;
        }
        return (int)(System.nanoTime() / 1000);
    }

    static long _mem() {
        Runtime rt = Runtime.getRuntime();
        rt.gc();
        return rt.totalMemory() - rt.freeMemory();
    }

    static boolean[] appendBool(boolean[] arr, boolean v) {
        boolean[] out = java.util.Arrays.copyOf(arr, arr.length + 1);
        out[arr.length] = v;
        return out;
    }

    static int _runeLen(String s) {
        return s.codePointCount(0, s.length());
    }

    static String _p(Object v) {
        if (v == null) return "<nil>";
        if (v.getClass().isArray()) {
            if (v instanceof int[]) return java.util.Arrays.toString((int[]) v);
            if (v instanceof long[]) return java.util.Arrays.toString((long[]) v);
            if (v instanceof double[]) return java.util.Arrays.toString((double[]) v);
            if (v instanceof boolean[]) return java.util.Arrays.toString((boolean[]) v);
            if (v instanceof byte[]) return java.util.Arrays.toString((byte[]) v);
            if (v instanceof char[]) return java.util.Arrays.toString((char[]) v);
            if (v instanceof short[]) return java.util.Arrays.toString((short[]) v);
            if (v instanceof float[]) return java.util.Arrays.toString((float[]) v);
            return java.util.Arrays.deepToString((Object[]) v);
        }
        return String.valueOf(v);
    }
}

File: tests/algorithms/x/PHP/strings/is_pangram.php
<?php
ini_set('memory_limit', '-1');
function _str($x) {
    if (is_array($x)) {
        $isList = array_keys($x) === range(0, count($x) - 1);
        if ($isList) {
            $parts = [];
            foreach ($x as $v) { $parts[] = _str($v); }
            return '[' . implode(' ', $parts) . ']';
        }
        $parts = [];
        foreach ($x as $k => $v) { $parts[] = _str($k) . ':' . _str($v); }
        return 'map[' . implode(' ', $parts) . ']';
    }
    if (is_bool($x)) return $x ? 'true' : 'false';
    if ($x === null) return 'null';
    return strval($x);
}
function _append($arr, $x) {
    $arr[] = $x;
    return $arr;
}
function _iadd($a, $b) {
    if (function_exists('bcadd')) {
        $sa = is_int($a) ? strval($a) : (is_string($a) ? $a : sprintf('%.0f', $a));
        $sb = is_int($b) ? strval($b) : (is_string($b) ? $b : sprintf('%.0f', $b));
        return bcadd($sa, $sb, 0);
    }
    return $a + $b;
}
function _isub($a, $b) {
    if (function_exists('bcsub')) {
        $sa = is_int($a) ? strval($a) : (is_string($a) ? $a : sprintf('%.0f', $a));
        $sb = is_int($b) ? strval($b) : (is_string($b) ? $b : sprintf('%.0f', $b));
        return bcsub($sa, $sb, 0);
    }
    return $a - $b;
}
function _imul($a, $b) {
    if (function_exists('bcmul')) {
        $sa = is_int($a) ? strval($a) : (is_string($a) ? $a : sprintf('%.0f', $a));
        $sb = is_int($b) ? strval($b) : (is_string($b) ? $b : sprintf('%.0f', $b));
        return bcmul($sa, $sb, 0);
    }
    return $a * $b;
}
function _idiv($a, $b) {
    return _intdiv($a, $b);
}
function _imod($a, $b) {
    if (function_exists('bcmod')) {
        $sa = is_int($a) ? strval($a) : (is_string($a) ? $a : sprintf('%.0f', $a));
        $sb = is_int($b) ? strval($b) : (is_string($b) ? $b : sprintf('%.0f', $b));
        return intval(bcmod($sa, $sb));
    }
    return $a % $b;
}
function is_pangram($input_str) {
  global $s1, $s2;
  $letters = [];
  $i = 0;
  while ($i < strlen($input_str)) {
  $c = strtolower(substr($input_str, $i, $i + 1 - $i));
  $is_new = !(in_array($c, $letters));
  if ($c != ' ' && 'a' <= $c && $c <= 'z' && $is_new) {
  $letters = _append($letters, $c);
}
  $i = _iadd($i, 1);
};
  return count($letters) == 26;
}
function is_pangram_faster($input_str) {
  global $s1, $s2;
  $alphabet = 'abcdefghijklmnopqrstuvwxyz';
  $flag = [];
  $i = 0;
  while ($i < 26) {
  $flag = _append($flag, false);
  $i = _iadd($i, 1);
};
  $j = 0;
  while ($j < strlen($input_str)) {
  $c = strtolower(substr($input_str, $j, $j + 1 - $j));
  $k = 0;
  while ($k < 26) {
  if (substr($alphabet, $k, $k + 1 - $k) == $c) {
  $flag[$k] = true;
  break;
}
  $k = _iadd($k, 1);
};
  $j = _iadd($j, 1);
};
  $t = 0;
  while ($t < 26) {
  if (!$flag[$t]) {
  return false;
}
  $t = _iadd($t, 1);
};
  return true;
}
function is_pangram_fastest($input_str) {
  global $s1, $s2;
  $s = strtolower($input_str);
  $alphabet = 'abcdefghijklmnopqrstuvwxyz';
  $i = 0;
  while ($i < strlen($alphabet)) {
  $letter = substr($alphabet, $i, $i + 1 - $i);
  if (!(strpos($s, $letter) !== false)) {
  return false;
}
  $i = _iadd($i, 1);
};
  return true;
}
$s1 = 'The quick brown fox jumps over the lazy dog';
$s2 = 'My name is Unknown';
echo rtrim(_str(is_pangram($s1))), PHP_EOL;
echo rtrim(_str(is_pangram($s2))), PHP_EOL;
echo rtrim(_str(is_pangram_faster($s1))), PHP_EOL;
echo rtrim(_str(is_pangram_faster($s2))), PHP_EOL;
echo rtrim(_str(is_pangram_fastest($s1))), PHP_EOL;
echo rtrim(_str(is_pangram_fastest($s2))), PHP_EOL;
